### DISCUSSION TITLE
Handle Curvilinear Discontiguities / Overlaps?

### DISCUSSION THREAD

**Discussion by trexfeathers** [SciTools/iris-esmf-regrid#195] [Ideas]
@jkettleb has asked about the long-term possibility of using iris-esmf-regrid on ORCA grids ([example](https://www.ncl.ucar.edu/Applications/Images/orca_2_2_lg.png)).

Since these can be loaded as Iris cubes with 2D coordinates, it seems likely that iris-esmf-regrid can already _attempt_ to process these grids via its curvilinear support, however the presence of discontiguities/overlaps could mean an invalid result being calculated. Such 'aberrations' are at least limited to terrestrial areas, so problems with the output might not actually matter, depending how they manifest. @stephenworsley is planning a cursory investigation.

If the output does prove unusable, there are two possible ways forward:

1. Improve iris-esmf-regrid to handle discontiguities/overlaps.
   Presumably will be complex work. We don't have full flexibility in this area given the use of ESMF as 
the engine.
2. Explore parsing ORCA grids via Iris' mesh data model.
   This has already been demonstrated for other layouts ([see docs page](https://scitools-iris.readthedocs.io/en/latest/further_topics/ugrid/other_meshes.html)), and would allow us to develop more efficiently. But we'd need to understand more about whether or not this would fit into user workflows.

**Comment by jkettleb** [comment]
thanks for looking into this

**Comment by zklaus** [comment]
The ORCA grids (which form a whole family) are massively simpler than for example the Voronoi grid employed by FESOM or the adaptively refined meshes. As such I strongly suggest to try and work this out with the normal curvilinear approach. As for discontinuities and overlaps, I think we can deal with the overlaps by cropping appropriate margins. This is the approach that Nemo itself has adopted starting from the most recent version 4.2, see [its migration guide](https://sites.nemo-ocean.io/user-guide/migration.html#migrating-from-4-0-x-to-4-2-x).

For existing data from CMIP5 and CMIP6, the situation is unfortunately a bit more complicated. Several ESMs/GCMs use Nemo as their ocean component, but some crop the output to remove the overlap, some mask it, and some keep the duplication. I think only one model, IPSL in CMIP5 (not in CMIP6!), has adopted the approach of cutting out parts of the domain and interleaving the output.

I suggest assuming and checking for output with no overlap and no weird interleaving. Does this still leave some discontinuities?

**Comment by stephenworsley** [comment]
I've done a little experimenting with ORCA grid files and it looks like they are in fact contiguous by Iris standards but with overlaps which seem to correspond with masked data. I think there's a good chance things should "just work" with a couple tweaks. I did uncover a bug in my experiments which caused failures when the target grid was curvilinear and the data was lazy, though I think this can be worked around by realising the data before regridding while I work on a fix.
I think there's still a question of understanding the effect of overlapping cells and how that could be mitigated by masking strategies. This is likely to be especially important when the target grid is an ORCA grid. One idea which could be more relevant here is for regridders to have an optional target mask so that problematic cells could be forced to be masked.

**Comment by trexfeathers** [comment]
@SciTools/peloton @stephenworsley is this still something we are interested in? If so should this be converted to an issue?

**Comment by trexfeathers** [comment]
Has this been solved by the below?
- #219

**Comment by pp-mo** [comment]
@scitools/peloton we're investigating need for this internally.
It sounds possible, but it's not clear at this point whether a real need exists.

**Comment by jkettleb** [comment]
I think there is a need to do area weighted regrid of orca data produced by the Met Office.

**Reply by ESadek-MO** [reply]
@SciTools/peloton We think this might already be possible in https://github.com/SciTools-incubator/iris-esmf-regrid/pull/219? Would you be able to confirm/ correct us on this?

**Reply by jkettleb** [reply]
would be great if it is.  Are you asking me or peleton (sorry, I'm not sure how things work).  If me then I'll try to get along to the next surgery to discuss.  I'm kind of busy, but would like to see if there is an easy path to help confirm.

**Reply by stephenworsley** [reply]
We use the peleton tag when we are answering on behalf of the peleton. We tend to have a different person leading the peleton each week so the idea is that this helps clarify the thread of conversation across peleton sessions.

Discussing this at a surgery sounds perfect.

**Comment by jkettleb** [comment]
[like]  Jamie Kettleborough reacted to your message:
________________________________
